common/djangoapps/entitlements/utils.py

```diff
@@ -0,0 +1,47 @@
+from course_modes.models import CourseMode
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from student.models import CourseEnrollment
+
+
+def is_course_run_entitlement_fullfillable(course_run_id, compare_date, entitlement):
+    """
+    Checks that the current run meets the following criteria for an entitlement
+
+    1) Are currently running or start in the future
+    2) A user can enroll in
+    3) A user can upgrade to the entitlement mode
+    4) Are not already currently enrolled in a session
+
+    Arguments:
+        course_run_id (String): The id of the Course run that is being checked.
+        compare_date: The date and time that we are comparing against.  Generally the current date and time.
+        user_enrolled_sessions (list): The list of sessions that the user is enrolled in currently.
+        mode (String): The current mode of the entitlement that we are verifying against.
+
+    Returns:
+        bool: True is the Course Run is fullfillable for the CourseEntitlement.
+    """
+    # Only courses that have not ended will be displayed
```

Comment:
Nit, move this comment down two lines above what it references (these might have been my initial issues, but might as well fix them here.)